src/test/java/de/doubleslash/usb_led_matrix/view/ConfigurationViewTest.java

```diff
@@ -124,39 +136,69 @@ void shouldLeaveTimeoutWhenColorChanges() {
    @Test
    void shouldTurnOffWhenCurrentTimeEqualsTimeout() {
       // Given
-      final LocalTime currentTime = LocalTime.of(21, 0);
+      final LocalTime currentTime = LocalTime.of(20, 0);
+      final LocalDate currentDate = LocalDate.now();
+      final LocalDateTime dateTime = LocalDateTime.of(currentDate, currentTime);
+
       configurationView.setUsbAdapter(usbAdapter);
       CommandLineOptions.setTimeoutInMinutes(60);
-      configurationView.timeLightsTurnedOffNow = LocalTime.of(20, 0);
+      configurationView.timeLightsTurnedOffNow = LocalTime.of(21, 0);
       // When
-      configurationView.turnOffAutomaticallyIfNeeded(currentTime);
+      configurationView.turnOffAutomaticallyIfNeeded(dateTime);
       // Then
-      verify(usbAdapter).updatePixel(Color.BLACK);
+      verify(usbAdapter, never()).updatePixel(Color.BLACK);
    }
 
    @Test
    void shouldNotTurnOffWhenCurrentTimeDoesNotReachTimeout() {
       // Given
       final LocalTime currentTime = LocalTime.of(19, 0);
+      final LocalDate currentDate = LocalDate.now();
+      final LocalDateTime dateTime = LocalDateTime.of(currentDate, currentTime);
+
       configurationView.setUsbAdapter(usbAdapter);
       CommandLineOptions.setTimeoutInMinutes(60);
       configurationView.timeLightsTurnedOffNow = LocalTime.of(20, 0);
       // When
-      configurationView.turnOffAutomaticallyIfNeeded(currentTime);
+      configurationView.turnOffAutomaticallyIfNeeded(dateTime);
       // Then
       verify(usbAdapter, never()).updatePixel(Color.BLACK);
    }
 
+   @Test
+   void shouldSwitchOffWhenTheCurrentTimeExceedsTheTimeoutAndANewDayHasStarted() {
+      // Given
+      final LocalDate currentDate = LocalDate.of(2024, 12, 23);
+      final LocalTime currentTime = LocalTime.of(23, 42, 39, 91);
+      final LocalDateTime dateTime = LocalDateTime.of(currentDate, currentTime);
+
+      configurationView.setUsbAdapter(usbAdapter);
+      CommandLineOptions.setTimeoutInMinutes(60);
+
+      LocalDateTime timeoutDateTime = LocalDateTime.of(currentDate.plusDays(1), LocalTime.of(0, 42, 33, 747380700));
+
+      configurationView.timeLightsTurnedOffNow = timeoutDateTime.toLocalTime();
+
+      // When
+      configurationView.turnOffAutomaticallyIfNeeded(dateTime);
+
+      // Then
+      verify(usbAdapter).updatePixel(Color.BLACK);
```

Comment:
I'm not sure about the test. As I understand the setup
- now is the 23.12.2024
- it should turn off on the 24.12.2024

and when you trigger this you assert that it should have turned off - besides the turn of date was not reached? Or did I understand the test wrong?